charm-slurmd/src/interface_slurmd.py

```diff
@@ -158,12 +183,19 @@ def _store_slurmctld_host_port(self, host, port):
         if port != self._stored.slurmctld_port:
             self._stored.slurmctld_port = port
 
-    def get_stored_munge_key(self):
-        """Retrieve the munge_key from the StoredState."""
-        return self._stored.munge_key
-
     def _store_slurmd_restart_uuid(self, restart_slurmd_uuid):
         self._stored.restart_slurmd_uuid = restart_slurmd_uuid
 
     def _get_slurmd_restart_uuid(self):
         return self._stored.restart_slurmd_uuid
+
+    def get_stored_munge_key(self):
+        """Retrieve the munge_key from the StoredState."""
+        return self._stored.munge_key
+
+    def configure_new_node(self):
+        """Set this node as not new and trigger a reconfiguration."""
+        inv = self.node_inventory
+        inv["new_node"] = False
+        self._relation.data[self.model.unit]['inventory'] = json.dumps(inv)
```

Comment:
```suggestion
        self.node_inventory = inv
```

We have the property (setter and getter) for `node_inventory`, is there a reason to not use here?